E0069 needs to be updated to new format

From: src/test/compile-fail/E0069.rs

Error E0069 needs a span_label, updating it from:

```
error[E0069]: `return;` in a function whose return type is not `()`
  --> src/test/compile-fail/E0069.rs:12:5
   |
12 |     return; //~ ERROR E0069
   |     ^^^^^^
```

To:

```
error[E0069]: `return;` in a function whose return type is not `()`
  --> src/test/compile-fail/E0069.rs:12:5
   |
12 |     return; //~ ERROR E0069
   |     ^^^^^^ return type is not ()
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -3383,8 +3383,10 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                             // FIXME(#32730) propagate obligations
                             .map(|InferOk { obligations, .. }| assert!(obligations.is_empty()));
                         if eq_result.is_err() {
-                            span_err!(tcx.sess, expr.span, E0069,
-                                      "`return;` in a function whose return type is not `()`");
+                            struct_span_err!(tcx.sess, expr.span, E0069,
+                                     "`return;` in a function whose return type is not `()`")
+                                .span_label(expr.span, &format!("return type is not ()"))
+                                .emit();
                         }
                     }
                 }

--- a/src/test/compile-fail/E0069.rs
+++ b/src/test/compile-fail/E0069.rs
@@ -9,7 +9,9 @@
 // except according to those terms.
 
 fn foo() -> u8 {
-    return; //~ ERROR E0069
+    return;
+    //~^ ERROR `return;` in a function whose return type is not `()`
+    //~| NOTE return type is not ()
 }
 
 fn main() {

